### PR TITLE
Khensler/vsan encryption in transit

### PR DESCRIPTION
**Pull request by khensler**
This PR closes #

The changes in this PR are as follows:

* Adds support for set and get vsan encryption in transit configuration

I have read the [contributor guidelines](CONTRIBUTING.md) and have completed the following:

* [X] **Formatted the code** using VSCode default formatter for PowerShell.
* [X] **Tested the code** end-to-end against an SDDC.
* [X] **Documented the functions** using standard PowerShell markup and applied `AVSAttribute` to newly exported functions.

### CODE REVIEW

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 [hunk 1]
```diff
@@ -2584,3 +2584,121 @@ Function Get-AVSVSANClusterUNMAPTRIM {
...
+Function Set-AVSVSANClusterEncryptionInTransit {
+    <#
+    .DESCRIPTION
+        This function enables vSAN Encryption in transit on the cluster defined by the -Name parameter.
+    .PARAMETER Name
+        Name of Clusters as defined in vCenter.  Valid values are blank or a comma separated list of cluster names.
+        Set-AVSVSANClusterEncryptionInTransit -Name Cluster-1,Cluster-2,Cluster-3
+        Enables vSAN Encryption in transit on Clusters-1,2,3
+        Set-AVSVSANClusterEncryptionInTransit -Enable:True
+        Enables vSAN Encryption in transit on all Clusters
```
Comment (et1975) [inline Microsoft.AVS.Management/Microsoft.AVS.Management.psm1:2598]: We don't have a way to show multi-line parameter description as a reasonably looking tooltip. Need to keep it singleline/short and move the examples to .EXAMPLE section (although we don't have a way to show that either).

**Review by et1975 — APPROVED** [review]

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 [hunk 2]
```diff
@@ -2584,3 +2584,121 @@ Function Get-AVSVSANClusterUNMAPTRIM {
...
+        This function gets vSAN Encryption in transit configuration status on all clusters.
+    #>
+
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    param ()
+    begin {}
+    process {
+            $Config = @()
+            $vSANConigView = Get-VsanView -Id VsanVcClusterConfigSystem-vsan-cluster-config-system
```
Comment (Zsoldier) [inline Microsoft.AVS.Management/Microsoft.AVS.Management.psm1:2694]: Same spelling.

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 [hunk 3]
```diff
@@ -2584,3 +2584,121 @@ Function Get-AVSVSANClusterUNMAPTRIM {
...
+                    }
+                }else {
+                    Write-Error "Failed to set $($Cluster.Name) to $Enable for vSAN Encryption in transit"
+                }
+            }
+        }
+        Else {
+            Foreach ($Entry in $Array) {
+                If ($Cluster = Get-Cluster -name $Entry) {
+                    $vSANConigView = Get-VsanView -Id VsanVcClusterConfigSystem-vsan-cluster-config-system
```
Comment (Zsoldier) [inline Microsoft.AVS.Management/Microsoft.AVS.Management.psm1:2658]: Variable spelling

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 (hunk 1 above)
Comment (Zsoldier) [reply Microsoft.AVS.Management/Microsoft.AVS.Management.psm1:2598]: Any way to add something like to the PR validation check?  Also maybe add to scripting guidelines?

**Review by Zsoldier — CHANGES_REQUESTED** [review]

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 (hunk 1 above)
Comment (khensler) [reply Microsoft.AVS.Management/Microsoft.AVS.Management.psm1]: Updated all to single line.

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 (hunk 2 above)
Comment (khensler) [reply Microsoft.AVS.Management/Microsoft.AVS.Management.psm1:2686]: updated spelling

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 (hunk 3 above)
Comment (khensler) [reply Microsoft.AVS.Management/Microsoft.AVS.Management.psm1]: updated spelling

File: Microsoft.AVS.Management/Microsoft.AVS.Management.psm1 [hunk 4]
```diff
@@ -2584,3 +2584,113 @@ Function Get-AVSVSANClusterUNMAPTRIM {
...
+                If ($Cluster = Get-Cluster -name $Entry) {
+                    $vSANConfigView = Get-VsanView -Id VsanVcClusterConfigSystem-vsan-cluster-config-system
+                    $vSANReconfigSpec = New-Object -type VMware.Vsan.Views.VimVsanReconfigSpec
+                    $vSANReconfigSpec.Modify = $true
+                    $vSANDataInTransitConfig= New-Object -type VMware.Vsan.Views.VsanDataInTransitEncryptionConfig
+                    $vSANDataInTransitConfig.Enabled = $Enable
+                    $vSANDataInTransitConfig.RekeyInterval = 1440
+                    $vSANReconfigSpec.DataInTransitEncryptionConfig = $vSANDataInTransitConfig
+                    $task = $vSANConfigView.VsanClusterReconfig($Cluster.ExtensionData.MoRef,$vSANReconfigSpec)
+                    Wait-Task -Task (Get-Task -Id $task)
```
Comment (et1975) [inline Microsoft.AVS.Management/Microsoft.AVS.Management.psm1:2658]: how long, roughly, are we waiting? The timeout for the command is enough at 10min?

Comment (khensler) [reply Microsoft.AVS.Management/Microsoft.AVS.Management.psm1:2658]: In my testing it's been only about 30 seconds.   But i haven't tested on a large cluster yet.

**Pull request closed without merging**